src/entities.rs

```diff
@@ -78,7 +84,14 @@ impl<'de> serde::Deserialize<'de> for Entity {
         D: serde::Deserializer<'de>,
     {
         let bits = u64::deserialize(deserializer)?;
-        Ok(Entity::from_bits(bits))
+
+        match Entity::from_bits(bits) {
+            Some(ent) => Ok(ent),
+            None => Err(serde::de::Error::invalid_value(
+                serde::de::Unexpected::Unsigned(bits),
+                &"`bits` does not correpond to possible `NonZeroU64` value ",
```

Comment:
This error text is incorrect, because many nonzero `u64`s still have zero for their upper 32 bits, making them invalid. You've also got a couple typos, and a user preforming serde deserialization is not aware of any variable named "bits." Finally, [the docs for `invalid_value`](https://docs.serde.rs/serde/de/trait.Error.html#method.invalid_value) indicate that the second argument should describe the expected value, not elaborate on what was actually found.

```suggestion
                &"a valid `Entity` bitpattern",
```